en/contributing/plugins.md

```diff
@@ -0,0 +1,102 @@
+# Writing Plugins
+
+DroneCore is split into a [core](https://github.com/dronecore/DroneCore/tree/master/core) and [plugins](https://github.com/dronecore/DroneCore/tree/master/plugins).
+
+Plugins that are located in the correct location (a subfolder of **/plugins**) and have the correct structure are included at compile time. The *cmake* script [autogenerate_plugin_container.cmake](https://github.com/dronecore/DroneCore/blob/master/autogenerate_plugin_container.cmake) takes care of including the plugin folders and integration tests.
+
+> **Note** Plugins can also be defined as part of an SDK/Extension Library (see [SDK Creation](../guide/sdk.md)).
+
+## Plugin Architecture
+
+Plugins should be written so that they are independent of each other (allowing them to be removed/replaced as needed).
+They will still need to be dependent on the core source, and there will be some duplicated functionality across the plugin modules.
+
+The code for each plugin and its unit test (if defined) is stored in a sub-folder of the **plugins** directory. Integration tests for all plugins are stored in **integration_tests**. 
+A simplified view of the folder structure is shown below: 
+
+```
+├── DroneCore
+│   ├── core
+│   ├── integration_tests
+│   └── plugins
+│       ├── action
+│       ├── ...
+│       └── telemetry
+```
+Drilling down, each plugin should have the same structure as shown for the "example" plugin below.
+```
+└── plugins
+    └── example 
+        ├── CMakeLists.txt
+        ├── example.cpp
+        ├── example.h
+        ├── example_impl.cpp
+        ├── example_impl.h
+        └── example_impl_test.cpp  ##optional
+```
+
+## Create a Plugin
+
+To create a new C++ plugin, copy the [example](https://github.com/dronecore/DroneCore/tree/master/external_example/plugins/example/) plugin from the "external_example" into the DroneCore **plugins** directory.
+
+> **Tip** Plugins are defined/written in the same way whether declared in DroneCore or 
+as part of an [SDK](../guide/sdk.md). The only difference is how they are included in the build.
+
+Modify the plugin as needed and update its [CMakeLists.txt](https://github.com/dronecore/DroneCore/blob/master/external_example/plugins/example/CMakeLists.txt) as appropriate:
+* Add additional libraries using the variable `additional_libs`:
+  ```
+  set(additional_libs "library_name" PARENT_SCOPE)
+  ```
+* Add required includes with `additional_includes`:
+  ```
+  set(additional_includes "include_dir" PARENT_SCOPE)
+  ```
+<!-- * You can also add tests with `unittest_source_files`, as discussed in the following section. -->

```

Comment:
This section was copied by me from the "SDK" plugin docs. I have commented out this line on `unittest_source_files` because it is not clear if it is used. I THINK it is, if present. Can you please confirm?

IF this stuff is exactly the same for both standard/external plugins I might just document it here and link from SDK docs.